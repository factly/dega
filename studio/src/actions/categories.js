import axios from 'axios';
import {
  ADD_CATEGORIES,
  ADD_CATEGORIES_REQUEST,
  SET_CATEGORIES_LOADING,
  RESET_CATEGORIES,
  CATEGORIES_API,
  GET_CATEGORY,
  UPDATE_CATEGORY,
} from '../constants/categories';
import { addErrorNotification, addSuccessNotification } from './notifications';
import { addMedia } from './media';
import getError from '../utils/getError';

// action to fetch all categories
export const getCategories = (query) => {
  return (dispatch, getState) => {
    const currentSpaceID = getState().spaces?.selected;
    if (currentSpaceID === 0) {
      return;
    }
    dispatch(loadingCategories());
    return axios
      .get(CATEGORIES_API, {
        params: query,
      })
      .then((response) => {
        dispatch(
          addMedia(
            response.data.nodes
              .filter((category) => category.medium)
              .map((category) => category.medium),
          ),
        );
        dispatch(
          addCategoriesList(
            response.data.nodes.map((category) => {
              return { ...category, medium: category.medium?.id };
            }),
          ),
        );
        dispatch(
          addCategoriesRequest({
            data: response.data.nodes.map((item) => item.id),
            query: query,
            total: response.data.total,
          }),
        );
      })
      .catch((error) => {
        dispatch(addErrorNotification(getError(error)));
      })
      .finally(() => dispatch(stopCategoriesLoading()));
  };
};

// action to fetch category by id
export const getCategory = (id) => {
  return (dispatch) => {
    dispatch(loadingCategories());
    return axios
      .get(CATEGORIES_API + '/' + id)
      .then((response) => {
        if (response.data.medium) dispatch(addMedia([response.data.medium]));

        dispatch(addCategory(GET_CATEGORY, { ...response.data, medium: response.data.medium?.id }));
      })
      .catch((error) => {
        dispatch(addErrorNotification(getError(error)));
      })
      .finally(() => dispatch(stopCategoriesLoading()));
  };
};

// action to create category
export const createCategory = (data) => {
  return (dispatch) => {
    dispatch(loadingCategories());
    return axios
      .post(CATEGORIES_API, data)
      .then(() => {
        dispatch(resetCategories());
        dispatch(addSuccessNotification('Category created'));
      })
      .catch((error) => {
        dispatch(addErrorNotification(getError(error)));
      });
  };
};

// action to update category by id
export const updateCategory = (data) => {
  return (dispatch) => {
    dispatch(loadingCategories());
    return axios
      .put(CATEGORIES_API + '/' + data.id, data)
      .then((response) => {
        if (response.data.medium) dispatch(addMedia([response.data.medium]));

        dispatch(
          addCategory(UPDATE_CATEGORY, { ...response.data, medium: response.data.medium?.id }),
        );
        dispatch(addSuccessNotification('Category updated'));
      })
      .catch((error) => {
        dispatch(addErrorNotification(getError(error)));
      })
      .finally(() => dispatch(stopCategoriesLoading()));
  };
};

// action to delete category by id
export const deleteCategory = (id) => {
  return (dispatch) => {
    dispatch(loadingCategories());
    return axios
      .delete(CATEGORIES_API + '/' + id)
      .then(() => {
        dispatch(resetCategories());
        dispatch(addSuccessNotification('Category deleted'));
      })
      .catch((error) => {
        dispatch(addErrorNotification(getError(error)));
      });
  };
};

export const addCategories = (categories) => {
  return (dispatch) => {
    dispatch(
      addMedia(categories.filter((category) => category.medium).map((category) => category.medium)),
    );
    dispatch(
      addCategoriesList(
        categories.map((category) => {
          return { ...category, medium: category.medium?.id };
        }),
      ),
    );
  };
};

export const loadingCategories = () => ({
  type: SET_CATEGORIES_LOADING,
  payload: true,
});

export const stopCategoriesLoading = () => ({
  type: SET_CATEGORIES_LOADING,
  payload: false,
});

export const addCategory = (type, payload) => ({
  type,
  payload,
});

export const addCategoriesList = (data) => ({
  type: ADD_CATEGORIES,
  payload: data,
});

export const addCategoriesRequest = (data) => ({
  type: ADD_CATEGORIES_REQUEST,
  payload: data,
});

export const resetCategories = () => ({
  type: RESET_CATEGORIES,
});
