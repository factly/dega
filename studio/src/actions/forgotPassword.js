const userId = localStorage.getItem('userId');
export const requestPasswordReset = async (userId) => {
  const response = await fetch(
    `${window.REACT_APP_ZITADEL_AUTHORITY}/v2/users/${userId}/password_reset`,
    {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
        Authorization: `Bearer ${window.REACT_APP_ZITADEL_PAT}`,
      },
      body: JSON.stringify({
        sendLink: {
          notificationType: 'NOTIFICATION_TYPE_Email',
        },
      }),
    },
  );

  if (!response.ok) {
    throw new Error('Failed to send reset email');
  }

  return response.json();
};

export const resetPassword = async (userId, newPassword, verificationCode) => {
  const response = await fetch(
    `${window.REACT_APP_ZITADEL_AUTHORITY}/v2/users/${userId}/password`,
    {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
        Authorization: `Bearer ${window.REACT_APP_ZITADEL_PAT}`,
      },
      body: JSON.stringify({
        newPassword: {
          password: newPassword,
          changeRequired: false,
        },
        verificationCode: verificationCode,
      }),
    },
  );

  if (!response.ok) {
    throw new Error('Failed to reset password');
  }

  return response.json();
};
