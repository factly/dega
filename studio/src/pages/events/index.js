import React from 'react';
import EventList from './components/EventList';
import { Space, Button, Row } from 'antd';
import { Link } from 'react-router-dom';
import { Helmet } from 'react-helmet';

function Events() {
  return (
    <Space direction="vertical">
      <Helmet title={'Events'} />
      <Row gutter={16} justify="end">
        <Link key="1" to="/admin/events/create">
          <Button type="primary">New Event</Button>
        </Link>
      </Row>
      <EventList />
    </Space>
  );
}

export default Events;
