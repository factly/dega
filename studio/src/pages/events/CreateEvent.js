import React from 'react';
import { useDispatch } from 'react-redux';
import { createEvent } from '../../actions/events';

import EventCreateForm from './components/EventForm';
import { Helmet } from 'react-helmet';
import useNavigation from '../../utils/useNavigation';

function CreateEvent() {
  const history = useNavigation();
  const dispatch = useDispatch();
  const onCreate = (values) => {
    dispatch(createEvent(values)).then(() => history('/admin/events'));
  };
  return (
    <>
      <Helmet title={'Create Event'} />
      <EventCreateForm onCreate={onCreate} />
    </>
  );
}

export default CreateEvent;
