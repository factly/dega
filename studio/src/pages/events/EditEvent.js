import React from 'react';
import EventEditForm from './components/EventForm';
import { useDispatch, useSelector } from 'react-redux';
import { updateEvent, getEvent } from '../../actions/events';

import { useParams } from 'react-router-dom';
import { Skeleton } from 'antd';
import RecordNotFound from '../../components/ErrorsAndImage/RecordNotFound';
import { Helmet } from 'react-helmet';
import useNavigation from '../../utils/useNavigation';

function EditEvent() {
  const history = useNavigation();
  const { id } = useParams();

  const dispatch = useDispatch();
  const { event, loading } = useSelector((state) => {
    return {
      event: state.events.details[id] ? state.events.details[id] : null,
      loading: state.events.loading,
    };
  });

  React.useEffect(() => {
    dispatch(getEvent(id));
  }, [dispatch, id]);

  if (loading) return <Skeleton />;

  if (!event) {
    return <RecordNotFound />;
  }

  const onUpdate = (values) => {
    dispatch(updateEvent({ ...event, ...values }));
    history(`/admin/events/${id}/edit`);
  };
  return (
    <>
      <Helmet title={'Edit Event'} />
      <EventEditForm data={event} onCreate={onUpdate} />
    </>
  );
}

export default EditEvent;
