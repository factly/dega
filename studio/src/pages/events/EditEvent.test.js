import React from 'react';
import { useHistory } from 'react-router-dom';
import { useDispatch, Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { act } from 'react-dom/test-utils';
import { mount } from 'enzyme';

import '../../matchMedia.mock';
import EditEvent from './EditEvent';
import * as actions from '../../actions/events';
import EventEditForm from './components/EventForm';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

jest.mock('react-monaco-editor', () => {
  const MonacoEditor = () => <div />;
  return MonacoEditor;
});
jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: jest.fn(),
}));

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: jest.fn(),
  useParams: jest.fn().mockReturnValue({ id: '1' }),
}));

jest.mock('../../actions/events', () => ({
  getEvents: jest.fn(),
  addEvent: jest.fn(),
  getEvent: jest.fn(),
  updateEvent: jest.fn(),
}));

let state = {
  events: {
    req: [],
    details: {},
    loading: true,
  },
  spaces: {
    orgs: [{ id: 1, organazation: 'Organization 1', spaces: [11] }],
    details: {
      11: { id: 11, name: 'Space 11' },
    },
    loading: false,
    selected: 11,
  },
};

describe('Event edit component', () => {
  let store;
  let mockedDispatch;
  store = mockStore(state);
  store.dispatch = jest.fn(() => ({}));
  mockedDispatch = jest.fn(() => Promise.resolve({}));
  useDispatch.mockReturnValue(mockedDispatch);

  describe('snapshot testing', () => {
    it('should render the component', () => {
      let tree;
      act(() => {
        tree = mount(
          <Provider store={store}>
            <EditEvent />
          </Provider>,
        );
      });
      expect(tree).toMatchSnapshot();
    });
    it('should match skeleton while loading', () => {
      let tree;
      store = mockStore({
        events: {
          req: [],
          details: {},
          loading: true,
        },
        spaces: {
          orgs: [{ id: 1, organazation: 'Organization 1', spaces: [11] }],
          details: {
            11: { id: 11, name: 'Space 11' },
          },
          loading: false,
          selected: 11,
        },
      });
      act(() => {
        tree = mount(
          <Provider store={store}>
            <Router>
              <EditEvent />
            </Router>
          </Provider>,
        );
      });
      expect(tree).toMatchSnapshot();
    });
  });
  describe('component testing', () => {
    let wrapper;
    state.events = {
      req: [],
      details: {
        1: {
          id: 1,
          name: 'event-1',
        },
      },
      loading: false,
    };
    beforeEach(() => {
      store = mockStore(state);
    });
    afterEach(() => {
      wrapper.unmount();
    });
    it('should call get action', () => {
      actions.getEvent.mockReset();
      act(() => {
        wrapper = mount(
          <Provider store={store}>
            <Router>
              <EditEvent />
            </Router>
          </Provider>,
        );
      });
      expect(actions.getEvent).toHaveBeenCalledWith('1');
    });
    it('should call updateEvent', (done) => {
      const push = jest.fn();
      useHistory.mockReturnValueOnce({ push });
      actions.updateEvent.mockReset();
      act(() => {
        wrapper = mount(
          <Provider store={store}>
            <Router>
              <EditEvent />
            </Router>
          </Provider>,
        );
      });

      wrapper
        .find(EventEditForm)
        .props()
        .onCreate({ tags: { app: 'dega' } });
      setTimeout(() => {
        expect(actions.updateEvent).toHaveBeenCalledWith({
          id: 1,
          name: 'event-1',
          tags: { app: 'dega' },
        });
        expect(push).toHaveBeenCalledWith('/admin/events/1/edit');
        done();
      }, 0);
    });
    it('should display RecordNotFound if event not found', () => {
      state.events = {
        req: [],
        details: {
          2: {
            id: 2,
            name: 'event-2',
          },
        },
        loading: false,
      };
      store = mockStore(state);
      act(() => {
        wrapper = mount(
          <Provider store={store}>
            <Router>
              <EditEvent />
            </Router>
          </Provider>,
        );
      });
      expect(wrapper.find('RecordNotFound').length).toBe(1);
      expect(wrapper.find(EventEditForm).length).toBe(0);
    });
  });
});
