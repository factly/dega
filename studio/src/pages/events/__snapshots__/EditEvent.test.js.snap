// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event edit component snapshot testing should match skeleton while loading 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <EditEvent>
        <Skeleton
          avatar={false}
          paragraph={true}
          title={true}
        >
          <div
            className="ant-skeleton"
          >
            <div
              className="ant-skeleton-content"
            >
              <Title
                prefixCls="ant-skeleton-title"
                width="38%"
              >
                <h3
                  className="ant-skeleton-title"
                  style={
                    Object {
                      "width": "38%",
                    }
                  }
                />
              </Title>
              <Paragraph
                prefixCls="ant-skeleton-paragraph"
                rows={3}
                width="61%"
              >
                <ul
                  className="ant-skeleton-paragraph"
                >
                  <li
                    key="0"
                    style={
                      Object {
                        "width": undefined,
                      }
                    }
                  />
                  <li
                    key="1"
                    style={
                      Object {
                        "width": undefined,
                      }
                    }
                  />
                  <li
                    key="2"
                    style={
                      Object {
                        "width": "61%",
                      }
                    }
                  />
                </ul>
              </Paragraph>
            </div>
          </div>
        </Skeleton>
      </EditEvent>
    </Router>
  </BrowserRouter>
</Provider>
`;

exports[`Event edit component snapshot testing should render the component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [MockFunction],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <EditEvent>
    <HelmetWrapper
      defer={true}
      encodeSpecialCharacters={true}
      title="Edit Event"
    >
      <SideEffect(NullComponent)
        defer={true}
        encodeSpecialCharacters={true}
        title="Edit Event"
      >
        <NullComponent
          defer={true}
          encodeSpecialCharacters={true}
          title="Edit Event"
        />
      </SideEffect(NullComponent)>
    </HelmetWrapper>
    <EventForm
      data={
        Object {
          "id": 1,
          "name": "event-1",
        }
      }
      onCreate={[Function]}
    >
      <ForwardRef(InternalForm)
        form={
          Object {
            "__INTERNAL__": Object {
              "itemRef": [Function],
              "name": "create-event",
            },
            "_init": true,
            "getFieldError": [Function],
            "getFieldInstance": [Function],
            "getFieldValue": [Function],
            "getFieldWarning": [Function],
            "getFieldsError": [Function],
            "getFieldsValue": [Function],
            "getInternalHooks": [Function],
            "isFieldTouched": [Function],
            "isFieldValidating": [Function],
            "isFieldsTouched": [Function],
            "isFieldsValidating": [Function],
            "resetFields": [Function],
            "scrollToField": [Function],
            "setFields": [Function],
            "setFieldsValue": [Function],
            "submit": [Function],
            "validateFields": [Function],
          }
        }
        initialValues={
          Object {
            "id": 1,
            "name": "event-1",
          }
        }
        labelCol={
          Object {
            "span": 9,
          }
        }
        name="create-event"
        onFinish={[Function]}
        wrapperCol={
          Object {
            "span": 7,
          }
        }
      >
        <SizeContextProvider>
          <ForwardRef(Form)
            className="ant-form ant-form-horizontal"
            form={
              Object {
                "__INTERNAL__": Object {
                  "itemRef": [Function],
                  "name": "create-event",
                },
                "_init": true,
                "getFieldError": [Function],
                "getFieldInstance": [Function],
                "getFieldValue": [Function],
                "getFieldWarning": [Function],
                "getFieldsError": [Function],
                "getFieldsValue": [Function],
                "getInternalHooks": [Function],
                "isFieldTouched": [Function],
                "isFieldValidating": [Function],
                "isFieldsTouched": [Function],
                "isFieldsValidating": [Function],
                "resetFields": [Function],
                "scrollToField": [Function],
                "setFields": [Function],
                "setFieldsValue": [Function],
                "submit": [Function],
                "validateFields": [Function],
              }
            }
            id="create-event"
            initialValues={
              Object {
                "id": 1,
                "name": "event-1",
              }
            }
            name="create-event"
            onFinish={[Function]}
            onFinishFailed={[Function]}
          >
            <form
              className="ant-form ant-form-horizontal"
              id="create-event"
              onReset={[Function]}
              onSubmit={[Function]}
            >
              <FormItem
                label="Name"
                name="name"
              >
                <WrapperField
                  label="Name"
                  messageVariables={
                    Object {
                      "label": "Name",
                    }
                  }
                  name="name"
                  onMetaChange={[Function]}
                  trigger="onChange"
                  validateTrigger="onChange"
                >
                  <Field
                    fieldContext={
                      Object {
                        "__INTERNAL__": Object {
                          "itemRef": [Function],
                          "name": "create-event",
                        },
                        "_init": true,
                        "getFieldError": [Function],
                        "getFieldInstance": [Function],
                        "getFieldValue": [Function],
                        "getFieldWarning": [Function],
                        "getFieldsError": [Function],
                        "getFieldsValue": [Function],
                        "getInternalHooks": [Function],
                        "isFieldTouched": [Function],
                        "isFieldValidating": [Function],
                        "isFieldsTouched": [Function],
                        "isFieldsValidating": [Function],
                        "resetFields": [Function],
                        "scrollToField": [Function],
                        "setFields": [Function],
                        "setFieldsValue": [Function],
                        "submit": [Function],
                        "validateFields": [Function],
                        "validateTrigger": "onChange",
                      }
                    }
                    key="_name"
                    label="Name"
                    messageVariables={
                      Object {
                        "label": "Name",
                      }
                    }
                    name={
                      Array [
                        "name",
                      ]
                    }
                    onMetaChange={[Function]}
                    trigger="onChange"
                    validateTrigger="onChange"
                    valuePropName="value"
                  >
                    <Row
                      className="ant-form-item"
                      key="row"
                    >
                      <div
                        className="ant-row ant-form-item"
                        role="row"
                        style={Object {}}
                      >
                        <FormItemLabel
                          htmlFor="create-event_name"
                          label="Name"
                          name="name"
                          prefixCls="ant-form"
                          required={false}
                          requiredMark={true}
                        >
                          <Col
                            className="ant-form-item-label"
                            span={9}
                          >
                            <div
                              className="ant-col ant-col-9 ant-form-item-label"
                              role="cell"
                              style={Object {}}
                            >
                              <label
                                className=""
                                htmlFor="create-event_name"
                                title="Name"
                              >
                                Name
                              </label>
                            </div>
                          </Col>
                        </FormItemLabel>
                        <FormItemInput
                          errors={Array []}
                          label="Name"
                          name={Array []}
                          prefixCls="ant-form"
                          status=""
                          touched={false}
                          validating={false}
                          warnings={Array []}
                        >
                          <Col
                            className="ant-form-item-control"
                            span={7}
                          >
                            <div
                              className="ant-col ant-col-7 ant-form-item-control"
                              role="cell"
                              style={Object {}}
                            >
                              <div
                                className="ant-form-item-control-input"
                              >
                                <div
                                  className="ant-form-item-control-input-content"
                                >
                                  <Component
                                    update={
                                      <ForwardRef
                                        placeholder="Enter name"
                                      />
                                    }
                                    value="event-1"
                                  >
                                    <ForwardRef
                                      id="create-event_name"
                                      onChange={[Function]}
                                      placeholder="Enter name"
                                      value="event-1"
                                    >
                                      <ForwardRef
                                        affixWrapperClassName=""
                                        groupClassName=""
                                        id="create-event_name"
                                        inputClassName=""
                                        onBlur={[Function]}
                                        onChange={[Function]}
                                        onFocus={[Function]}
                                        placeholder="Enter name"
                                        prefixCls="ant-input"
                                        value="event-1"
                                        wrapperClassName=""
                                      >
                                        <BaseInput
                                          affixWrapperClassName=""
                                          focused={false}
                                          groupClassName=""
                                          handleReset={[Function]}
                                          id="create-event_name"
                                          inputElement={
                                            <input
                                              className="ant-input"
                                              id="create-event_name"
                                              onBlur={[Function]}
                                              onChange={[Function]}
                                              onFocus={[Function]}
                                              onKeyDown={[Function]}
                                              placeholder="Enter name"
                                              type="text"
                                              value="event-1"
                                            />
                                          }
                                          placeholder="Enter name"
                                          prefixCls="ant-input"
                                          suffix={null}
                                          triggerFocus={[Function]}
                                          value="event-1"
                                          wrapperClassName=""
                                        >
                                          <input
                                            className="ant-input"
                                            id="create-event_name"
                                            onBlur={[Function]}
                                            onChange={[Function]}
                                            onFocus={[Function]}
                                            onKeyDown={[Function]}
                                            placeholder="Enter name"
                                            type="text"
                                            value="event-1"
                                          />
                                        </BaseInput>
                                      </ForwardRef>
                                    </ForwardRef>
                                  </Component>
                                </div>
                              </div>
                              <ErrorList
                                className="ant-form-item-explain-connected"
                                errors={Array []}
                                helpStatus=""
                                warnings={Array []}
                              >
                                <CSSMotion
                                  motionAppear={false}
                                  motionDeadline={500}
                                  motionEnter={false}
                                  motionName="ant-show-help"
                                  onAppearActive={[Function]}
                                  onAppearEnd={[Function]}
                                  onAppearStart={[Function]}
                                  onEnterActive={[Function]}
                                  onEnterEnd={[Function]}
                                  onEnterStart={[Function]}
                                  onLeaveActive={[Function]}
                                  onLeaveEnd={[Function]}
                                  onLeaveStart={[Function]}
                                  visible={false}
                                >
                                  <DomWrapper />
                                </CSSMotion>
                              </ErrorList>
                            </div>
                          </Col>
                        </FormItemInput>
                      </div>
                    </Row>
                  </Field>
                </WrapperField>
              </FormItem>
              <FormItem
                label="Tags"
                name="tags"
              >
                <WrapperField
                  label="Tags"
                  messageVariables={
                    Object {
                      "label": "Tags",
                    }
                  }
                  name="tags"
                  onMetaChange={[Function]}
                  trigger="onChange"
                  validateTrigger="onChange"
                >
                  <Field
                    fieldContext={
                      Object {
                        "__INTERNAL__": Object {
                          "itemRef": [Function],
                          "name": "create-event",
                        },
                        "_init": true,
                        "getFieldError": [Function],
                        "getFieldInstance": [Function],
                        "getFieldValue": [Function],
                        "getFieldWarning": [Function],
                        "getFieldsError": [Function],
                        "getFieldsValue": [Function],
                        "getInternalHooks": [Function],
                        "isFieldTouched": [Function],
                        "isFieldValidating": [Function],
                        "isFieldsTouched": [Function],
                        "isFieldsValidating": [Function],
                        "resetFields": [Function],
                        "scrollToField": [Function],
                        "setFields": [Function],
                        "setFieldsValue": [Function],
                        "submit": [Function],
                        "validateFields": [Function],
                        "validateTrigger": "onChange",
                      }
                    }
                    key="_tags"
                    label="Tags"
                    messageVariables={
                      Object {
                        "label": "Tags",
                      }
                    }
                    name={
                      Array [
                        "tags",
                      ]
                    }
                    onMetaChange={[Function]}
                    trigger="onChange"
                    validateTrigger="onChange"
                    valuePropName="value"
                  >
                    <Row
                      className="ant-form-item"
                      key="row"
                    >
                      <div
                        className="ant-row ant-form-item"
                        role="row"
                        style={Object {}}
                      >
                        <FormItemLabel
                          htmlFor="create-event_tags"
                          label="Tags"
                          name="tags"
                          prefixCls="ant-form"
                          required={false}
                          requiredMark={true}
                        >
                          <Col
                            className="ant-form-item-label"
                            span={9}
                          >
                            <div
                              className="ant-col ant-col-9 ant-form-item-label"
                              role="cell"
                              style={Object {}}
                            >
                              <label
                                className=""
                                htmlFor="create-event_tags"
                                title="Tags"
                              >
                                Tags
                              </label>
                            </div>
                          </Col>
                        </FormItemLabel>
                        <FormItemInput
                          errors={Array []}
                          label="Tags"
                          name={Array []}
                          prefixCls="ant-form"
                          status=""
                          touched={false}
                          validating={false}
                          warnings={Array []}
                        >
                          <Col
                            className="ant-form-item-control"
                            span={7}
                          >
                            <div
                              className="ant-col ant-col-7 ant-form-item-control"
                              role="cell"
                              style={Object {}}
                            >
                              <div
                                className="ant-form-item-control-input"
                              >
                                <div
                                  className="ant-form-item-control-input-content"
                                >
                                  <Component
                                    update={
                                      <MONACOEditor
                                        language="json"
                                      />
                                    }
                                  >
                                    <MONACOEditor
                                      id="create-event_tags"
                                      language="json"
                                      onChange={[Function]}
                                    >
                                      <MonacoEditor
                                        height="240"
                                        language="json"
                                        onChange={[Function]}
                                        options={
                                          Object {
                                            "autoClosingBrackets": "auto",
                                            "autoClosingQuotes": "auto",
                                            "cursorBlinking": "smooth",
                                            "folding": true,
                                            "lineNumbersMinChars": 4,
                                            "minimap": Object {
                                              "enabled": false,
                                            },
                                            "scrollBeyondLastLine": false,
                                            "scrollbar": Object {
                                              "alwaysConsumeMouseWheel": false,
                                            },
                                            "theme": "vs-dark",
                                            "wordWrap": "on",
                                          }
                                        }
                                        width={440}
                                      >
                                        <div />
                                      </MonacoEditor>
                                    </MONACOEditor>
                                  </Component>
                                </div>
                              </div>
                              <ErrorList
                                className="ant-form-item-explain-connected"
                                errors={Array []}
                                helpStatus=""
                                warnings={Array []}
                              >
                                <CSSMotion
                                  motionAppear={false}
                                  motionDeadline={500}
                                  motionEnter={false}
                                  motionName="ant-show-help"
                                  onAppearActive={[Function]}
                                  onAppearEnd={[Function]}
                                  onAppearStart={[Function]}
                                  onEnterActive={[Function]}
                                  onEnterEnd={[Function]}
                                  onEnterStart={[Function]}
                                  onLeaveActive={[Function]}
                                  onLeaveEnd={[Function]}
                                  onLeaveStart={[Function]}
                                  visible={false}
                                >
                                  <DomWrapper />
                                </CSSMotion>
                              </ErrorList>
                            </div>
                          </Col>
                        </FormItemInput>
                      </div>
                    </Row>
                  </Field>
                </WrapperField>
              </FormItem>
              <FormItem
                wrapperCol={
                  Object {
                    "offset": 10,
                    "span": 14,
                  }
                }
              >
                <Row
                  className="ant-form-item"
                  key="row"
                >
                  <div
                    className="ant-row ant-form-item"
                    role="row"
                    style={Object {}}
                  >
                    <FormItemLabel
                      prefixCls="ant-form"
                      requiredMark={true}
                      wrapperCol={
                        Object {
                          "offset": 10,
                          "span": 14,
                        }
                      }
                    />
                    <FormItemInput
                      errors={Array []}
                      name={Array []}
                      prefixCls="ant-form"
                      status=""
                      touched={false}
                      validating={false}
                      warnings={Array []}
                      wrapperCol={
                        Object {
                          "offset": 10,
                          "span": 14,
                        }
                      }
                    >
                      <Col
                        className="ant-form-item-control"
                        offset={10}
                        span={14}
                      >
                        <div
                          className="ant-col ant-col-14 ant-col-offset-10 ant-form-item-control"
                          role="cell"
                          style={Object {}}
                        >
                          <div
                            className="ant-form-item-control-input"
                          >
                            <div
                              className="ant-form-item-control-input-content"
                            >
                              <Space>
                                <div
                                  className="ant-space ant-space-horizontal ant-space-align-center"
                                  style={Object {}}
                                >
                                  <Item
                                    className="ant-space-item"
                                    direction="horizontal"
                                    index={0}
                                    key="ant-space-item-0"
                                    marginDirection="marginRight"
                                    wrap={false}
                                  >
                                    <div
                                      className="ant-space-item"
                                      style={
                                        Object {
                                          "marginRight": 8,
                                        }
                                      }
                                    >
                                      <Button
                                        htmlType="submit"
                                        type="primary"
                                      >
                                        <Wave
                                          disabled={false}
                                        >
                                          <button
                                            className="ant-btn ant-btn-primary"
                                            onClick={[Function]}
                                            type="submit"
                                          >
                                            <LoadingIcon
                                              existIcon={false}
                                              loading={false}
                                              prefixCls="ant-btn"
                                            >
                                              <CSSMotion
                                                motionName="ant-btn-loading-icon-motion"
                                                onAppearActive={[Function]}
                                                onAppearStart={[Function]}
                                                onEnterActive={[Function]}
                                                onEnterStart={[Function]}
                                                onLeaveActive={[Function]}
                                                onLeaveStart={[Function]}
                                                removeOnLeave={true}
                                                visible={false}
                                              >
                                                <DomWrapper />
                                              </CSSMotion>
                                            </LoadingIcon>
                                            <span
                                              key=".0"
                                            >
                                              Submit
                                            </span>
                                          </button>
                                        </Wave>
                                      </Button>
                                    </div>
                                  </Item>
                                  <Item
                                    className="ant-space-item"
                                    direction="horizontal"
                                    index={1}
                                    key="ant-space-item-1"
                                    marginDirection="marginRight"
                                    wrap={false}
                                  >
                                    <div
                                      className="ant-space-item"
                                      style={Object {}}
                                    >
                                      <Button
                                        htmlType="button"
                                        onClick={[Function]}
                                      >
                                        <Wave
                                          disabled={false}
                                        >
                                          <button
                                            className="ant-btn ant-btn-default"
                                            onClick={[Function]}
                                            type="button"
                                          >
                                            <LoadingIcon
                                              existIcon={false}
                                              loading={false}
                                              prefixCls="ant-btn"
                                            >
                                              <CSSMotion
                                                motionName="ant-btn-loading-icon-motion"
                                                onAppearActive={[Function]}
                                                onAppearStart={[Function]}
                                                onEnterActive={[Function]}
                                                onEnterStart={[Function]}
                                                onLeaveActive={[Function]}
                                                onLeaveStart={[Function]}
                                                removeOnLeave={true}
                                                visible={false}
                                              >
                                                <DomWrapper />
                                              </CSSMotion>
                                            </LoadingIcon>
                                            <span
                                              key=".0"
                                            >
                                              Reset
                                            </span>
                                          </button>
                                        </Wave>
                                      </Button>
                                    </div>
                                  </Item>
                                </div>
                              </Space>
                            </div>
                          </div>
                          <ErrorList
                            className="ant-form-item-explain-connected"
                            errors={Array []}
                            helpStatus=""
                            warnings={Array []}
                          >
                            <CSSMotion
                              motionAppear={false}
                              motionDeadline={500}
                              motionEnter={false}
                              motionName="ant-show-help"
                              onAppearActive={[Function]}
                              onAppearEnd={[Function]}
                              onAppearStart={[Function]}
                              onEnterActive={[Function]}
                              onEnterEnd={[Function]}
                              onEnterStart={[Function]}
                              onLeaveActive={[Function]}
                              onLeaveEnd={[Function]}
                              onLeaveStart={[Function]}
                              visible={false}
                            >
                              <DomWrapper />
                            </CSSMotion>
                          </ErrorList>
                        </div>
                      </Col>
                    </FormItemInput>
                  </div>
                </Row>
              </FormItem>
            </form>
          </ForwardRef(Form)>
        </SizeContextProvider>
      </ForwardRef(InternalForm)>
    </EventForm>
  </EditEvent>
</Provider>
`;
